### ISSUE TITLE
Group the modules by package in the sidebar

### ISSUE THREAD

**Issue by f-f** [spacchetti/purescript-docs-search#20]
Feature request from Slack:

<img width="772" alt="image" src="https://user-images.githubusercontent.com/5480361/64361714-676f2400-d00d-11e9-818d-40d39b2bfc24.png">

We could fold and hide the modules list for every package name, and show it in a nested list once the user clicks on some arrow

**Comment by m-bock** [comment]
Yeah, I definitely think that it would be nice to somehow group the modules in the docs by the package they're coming from. Here are some ideas:

- The modules could be already grouped in the folder structure. So you'd have paths like `generated-docs/html/control/Control.Alt.html` or why not even exactly like on pursuit `purescript-control/4.2.0/docs/Control.Alt`
- The actual `generated-docs/index.html` would be then just like the package overview on https://pursuit.purescript.org/ but limited to the packages used in your project.

Having this included in the sidebar may be some feature on top of this. It brings in further complexity (how should the UI look like, why differ from pursuit etc.)

**Comment by f-f** [comment]
> The modules could be already grouped in the folder structure. So you'd have paths like `generated-docs/html/control/Control.Alt.html` or why not even exactly like on pursuit `purescript-control/4.2.0/docs/Control.Alt`

The work happening here is doing nothing more than taking the output of `purs docs` (which generates all the html), generating the search index, and injecting some JS in the generated pages to integrate the search in the index.

So in light of this I wouldn't like to mess with the folder structure that is generated by `purs`, because whatever we do then we'd have to take extra care to not break all the links, etc. In the end I don't think it's worth if the result is just to get a nicer folder structure.

> The actual `generated-docs/index.html` would be then just like the package overview on pursuit.purescript.org but limited to the packages used in your project.

Considering the above (i.e. that here we do not generate the whole html, but just patch the existing one), I'd really like to have such an overview, yeah - the current page is quite overwhelming and not that useful IMHO

**Comment by athanclark** [comment]
Hi everyone,

I made a janky [monkey-patched script](https://github.com/openchronology/openchronology.github.io/blob/99d97432b3450c72c50cd126a9e59f07e4333004/src/Scripts/TrimDocs.purs) to help with this - it does the job by binding with [cheerio](https://www.npmjs.com/package/cheerio) (the PureScript bindings didn't really look up-to-par, so I quickly made my own in that script).

Although it would probably be in the interest of Spago to use HXT from haskell, like how pursuit does (I think?), it might be a temporary solution for some. Also, you might have to change line 27 and 32; I moved `generated-docs/html/` to just `generated-docs/`.
